core/src/main/scala/org/apache/spark/util/ListenerBus.scala

```diff
@@ -150,4 +152,12 @@ private[spark] trait ListenerBus[L <: AnyRef, E] extends Logging {
     listeners.asScala.filter(_.getClass == c).map(_.asInstanceOf[T]).toSeq
   }
 
+  private[spark] def redactEvent(e: E): E = {
```

Comment:
Since this is designed to be used in `ListenerBus` only, `private` will be enough in this case. For the other classes, this PR makes `EventLoggingListener.redactEvent` available already.